### DISCUSSION TITLE
Can obscured script run in cygwin

### DISCUSSION THREAD

**Discussion by shaobo-zhang-senscomm** [dashingsoft/pyarmor#1784] [General]
i obscure script with “--platform windows.x86_64 --platform linux.x86_64 --platform darwin.x86_64” parameter. But the obscured script can't run in cygwin. Windows and Linux envs are ok.

**Comment by shaobo-zhang-senscomm** [comment]
```
  File "/home/shaobo.zhang/scm-auto-obscure/CI/pyarmor_runtime_006272/__init__.py", line 49, in __pyarmor__
    return __import__(name, globals(), locals(), ['__pyarmor__'], level=1)
ModuleNotFoundError: No module named 'pyarmor_runtime_006272.py39.windows_x86_64.pyarmor_runtime'
```


```
shaobo.zhang@PC-SSZ060-1 ~/scm-auto-obscure/CI/pyarmor_runtime_006272/py39/windows_x86_64
$ ls
pyarmor_runtime.pyd
```
from __init__.py file in pyarmor_runtime_xxx, it should support cygwin.

**Comment by jondy** [comment]
It should support cygwin. Check all the dependent by this command
```
cygcheck /path/to/pyarmor_runtime.pyd
```

**Comment by jondy** [comment]
Also make sure cygwin python is Python 3.9

**Comment by shaobo-zhang-senscomm** [comment]
Hi @jondy 
```
$ cygcheck pyarmor_runtime_006272/py39/windows_x86_64/pyarmor_runtime.pyd
C:\cygwin64\home\shaobo.zhang\scm-auto-obscure\CI\pyarmor_runtime_006272\py39\windows_x86_64\pyarmor_runtime.pyd
cygcheck: track_down: could not find python39.dll
```

The lastest python version on cygwin is 3.9.16. After installation, there is no python39.dll file. And the lastest build file of 3.9 is 3.9.13 (no build files for 3.9.14 to 3.9.16), I copy python39.dll from 3.9.13 to /usr/bin in cygwin. 

cygcheck is ok. But pyarmor_runtime import still fail.

```
$ cygcheck pyarmor_runtime_006272/py39/windows_x86_64/pyarmor_runtime.pyd
C:\cygwin64\home\shaobo.zhang\scm-auto-obscure\CI\pyarmor_runtime_006272\py39\windows_x86_64\pyarmor_runtime.pyd
  C:\cygwin64\bin\python39.dll
```

```
    return __import__(name, globals(), locals(), ['__pyarmor__'], level=1)
ModuleNotFoundError: No module named 'pyarmor_runtime_006272.py39.windows_x86_64.pyarmor_runtime'
```

**Comment by jondy** [comment]
Copy `python39.dll` doesn't work, it's Windows DLL, not for cygwin.
It seems this `pyarmor_runtime.pyd` could not be used in cygwin directly.
One solution is that using some tool to patch `pyarmor_runtime.pyd`, change the depenedent `python39.dll` to `libpython39.dll` or `libpython39m.dll`
The other solution is to rebuild `pyarmor_runtime.pyd`

**Comment by shaobo-zhang-senscomm** [comment]
Hi @jondy,
Thanks for you reply. 
How to fix it. Is there a document?

**Comment by jondy** [comment]
Is there any `libpython39*.dll`? Try to create a link `python39.dll` to `libpython39*.dll`
